build-support/bin/ci.sh

```diff
@@ -107,13 +106,13 @@ export PANTS_DEV=1
 
 # Determine which Python interpreter to use for bootstrapping pex and for executing subprocesses.
 # Order matters here. We must constrain subprocesses before running the bootstrap stage,
-# or we will encounter the _Py_Dealloc error when bootstrapping a Python 3 PEX.
+# or we will encounter the _Py_Dealloc error when bootstrapping a Python 3 PEX (#6985).
 if [[ "${python_two:-false}" == "false" ]]; then
-  py_version_number="3"
-  bootstrap_pants_script="./pants3"
-  export PANTS_PYTHON_SETUP_INTERPRETER_CONSTRAINTS='["CPython>=3.6,<4"]'
+  py_version_number="3.6"
+  bootstrap_pants_script="PY=python3.6 ./pants3"
+  export PANTS_PYTHON_SETUP_INTERPRETER_CONSTRAINTS='["CPython==3.6.*"]'
```

Comment:
This export can be killed now.

Reply:
We have to keep it for CI shards that pull down `./pants.pex` from AWS, as they not have interpreter constraints set so will try to use Py2 for subprocesses, leading to `_Py_Dealloc`.

I'll update the comments to reflect this. Looks like they've fallen out of date.